server/backend/api/app/routers/router_guards.py

```diff
@@ -235,35 +235,43 @@ async def authorize_by_privilege(
 
     # if requester is not application admin, check if has the privilege to grant/remove access of the role
     if not requester_is_app_admin:
-        # for remove access, role is what we get above from Depends(get_request_role_from_id)
-        # for grant access, if the request params has forest_client_number, need to get the child role
-        # the child role should already exist when the delegated admin has been created
         if "user_role_xref_id" not in request.path_params:
+            # for grant access with abstract role, if the request params has forest_client_numbers, need to get the child role
+            # the child role should already exist when the delegated admin has been created
             rbody = await request.json()
-            forest_client_number = rbody.get("forest_client_number")
+            forest_client_numbers = rbody.get("forest_client_numbers")
             if (
-                forest_client_number
+                forest_client_numbers
                 and role.role_type_code == RoleType.ROLE_TYPE_ABSTRACT
             ):
                 parent_role = role
-                forest_client_role_name = (
-                    crud_user_role.construct_forest_client_role_name(
-                        parent_role.role_name, forest_client_number
+                for forest_client_number in forest_client_numbers:
```

Comment:
Hi Ian, when we accept multiple forest client numbers from frontend, for delegated admin, we need to check the privilege for each forest client number. So before making too many changes, for now, as long as the delegated admin miss the privilege for one of the forest client numbers, will throw permission error. We could discuss that in our scrum next week. I'm thinking in the future, once we only allow delegated admin to select forest client numbers from allowed list, this won't be a big problem.